opengemini/query_builder.go

```diff
@@ -0,0 +1,184 @@
+package opengemini
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type QueryBuilder struct {
+	selectExprs []Expression
+	from        []string
+	where       Condition
+	groupBy     []Expression
+	order       SortOrder
+	limit       int64
+	offset      int64
+	timezone    *time.Location
+}
+
+func CreateQueryBuilder() *QueryBuilder {
+	return &QueryBuilder{}
+}
+
+func (q *QueryBuilder) Select(selectExpressions ...Expression) *QueryBuilder {
+	q.selectExprs = selectExpressions
+	return q
+}
+
+func (q *QueryBuilder) From(tables ...string) *QueryBuilder {
+	q.from = tables
+	return q
+}
+
+func (q *QueryBuilder) Where(condition Condition) *QueryBuilder {
+	q.where = condition
+	return q
+}
+
+func (q *QueryBuilder) GroupBy(groupByExpressions ...Expression) *QueryBuilder {
+	q.groupBy = groupByExpressions
+	return q
+}
+
+func (q *QueryBuilder) OrderBy(order SortOrder) *QueryBuilder {
+	q.order = order
+	return q
+}
+
+func (q *QueryBuilder) Limit(limit int64) *QueryBuilder {
+	q.limit = limit
+	return q
+}
+
+func (q *QueryBuilder) Offset(offset int64) *QueryBuilder {
+	q.offset = offset
+	return q
+}
+
+func (q *QueryBuilder) Timezone(location *time.Location) *QueryBuilder {
+	q.timezone = location
+	return q
+}
+
+func (q *QueryBuilder) Build() *Query {
+	var commandBuilder strings.Builder
```

Comment:
Based on the usage scenario, whether to define strings.builder as a member variable and reset it each time the builder is used.

Reply:
It's more like QueryTemplate? I think we didn't have highly performance requirements on QL 's concatenation. I think it's ok to keep this.